cmd/nse/nse.go

```diff
@@ -0,0 +1,171 @@
+		}),
+	)
+
+	return c, err
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
+
+	// Checking if nsm client socket exists and of not crash NSE
+	clientSocket := clientSocketPath
+	if clientSocketUserPath != nil {
+		clientSocket = *clientSocketUserPath
+	}
+
+	if _, err := os.Stat(clientSocket); err != nil {
+		logrus.Errorf("nse: failure to access nsm socket at %s with error: %+v, exiting...", clientSocket, err)
+		os.Exit(1)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), clientConnectionTimeout)
+	defer cancel()
+	conn, err := dial(ctx, clientSocket)
+	if err != nil {
+		logrus.Errorf("nse: failure to communicate with the socket %s with error: %+v", clientSocket, err)
+		os.Exit(1)
+	}
+	defer conn.Close()
+	logrus.Infof("nse: connection to nsm server on socket: %s succeeded.", clientSocket)
+
+	// NSM socket path will be used to drop NSE socket for NSM's Connection request
+	nsePath, _ := filepath.Split(clientSocket)
+	if err := socketCleanup(path.Join(nsePath, *nseSocketName)); err != nil {
+		logrus.Errorf("nse: failure to cleanup stale sicket %s with error: %+v", path.Join(nsePath, *nseSocketName), err)
+		os.Exit(1)
+	}
+	nse, err := net.Listen("unix", path.Join(nsePath, *nseSocketName))
+	grpcServer := grpc.NewServer([]grpc.ServerOption{}...)
+
+	go func() {
+		if err := grpcServer.Serve(nse); err != nil {
+			logrus.Errorf("nse: failed to start grpc server on socket %s with error: %+v ", path.Join(nsePath, *nseSocketName), err)
+		}
+	}()
+	// Check if the socket of device plugin server is operation
+	if err := socketOperationCheck(path.Join(nsePath, *nseSocketName)); err != nil {
+		logrus.Errorf("nse: failure to communicate with the socket %s with error: %+v", path.Join(nsePath, *nseSocketName), err)
+		os.Exit(1)
+	}
+
+	// Ok, NSE server is ready and now the channel can be advertised to NSM
+	nsmClient := nsmconnect.NewClientConnectionClient(conn)
+
+	channel := netmesh.NetworkServiceChannel{
+		Metadata: &common.Metadata{
+			Name: "Channel-1",
+		},
+		NetworkServiceName: "networkservice-1",
+		Payload:            "ipv4",
+		SocketLocation:     path.Join(nsePath, *nseSocketName),
+		Interface: []*common.Interface{
+			{
+				Type: common.InterfaceType_KERNEL_INTERFACE,
+				Metadata: &common.Metadata{
+					Name: "kernel_interface_1",
+				},
+				Preference: common.InterfacePreference_FIRST,
+			},
+		},
+	}
+	channels := []*netmesh.NetworkServiceChannel{}
+	channels = append(channels, &channel)
+	logrus.Infof("Channel info: %+v", channels)
+	resp, err := nsmClient.RequestAdvertiseChannel(context.Background(), &nsmconnect.ChannelAdvertiseRequest{
+		NetmeshChannel: channels,
+	})
+	if err != nil {
+		grpcServer.Stop()
+		logrus.Errorf("nse: failure to communicate with the socket %s with error: %+v", clientSocket, err)
+		os.Exit(1)
+
+	}
+	if !resp.Success {
+		grpcServer.Stop()
+		logrus.Error("nse: NSM response is inidcating failure of accepting Channel Advertisiment.")
```

Comment:
* Rename inidcating to indicating.
* Rename Advertisiment to Advertisement
* Use log.Fatalln instead of following up with os.Exit

Also please consider using lower cases for error messages as much as possible to make searching easier for operators.